Updating packages #2

jina2k opened the pull request:
# Description

Partially Addresses #548 ,
#662 and #463 

npm audit fix to resolve new commits
Packages changed:
### babel v6 to v7.
.babelrc -> babel.config.json with new params due to new @babel/core

babel params in package.json as well

babel-plugin-transform-object-rest-spread -> @babel/plugin-proposal-object-rest-spread
babel-preset-react -> @babel/preset-react
babel-plugin-react-intl -> babel-plugin-formatjs
babel-preset-env -> @babel/preset-env
react-dropzone update
attr-accept
source-map-resolve
@babel/cli
@babel/core
babel-core bridge

### Transition to ts-loader from awesome-typescript-loader

https://github.com/s-panferov/awesome-typescript-loader
as said in documentation here, awesome-typescript-loader aims to support only typescript@2.x and webpack@2.x.
As such, we will use the recommended ts-loader instead, even though there will be performance degradation.
This is mentioned in https://webpack.js.org/guides/typescript/

I've also fixed some glaring issues revolving around syntax errors that somehow made it to the build (only made obvious after transition from commonjs to es6), files listed are: app/types/redux/actions.ts, app/actions/admin.ts, app/actions/groups.ts

### React-intl v2 to v5

react-intlv2->react-intlv5.21.0
@formatjs/intl

## Type of change

- [x] Note merging this changes the node modules
- [ ] Note merging this changes the database configuration.
- [x] This change requires a documentation update

## Checklist

- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [x] I have removed text in ( ) from the issue request

## Limitations
Cannot upgrade to typescript v4.5.2 without changing the script test case as well as declaring types in multiple areas of the code. https://github.com/formatjs/formatjs/issues/3276
I attempted to use better-typescript-lib but found it easier to revert to 4.4.4 as this is not a new project.

jina2k committed:
removed stub types

jina2k committed:
babel v6 to v7

jina2k committed:
ts-loader and fixed the syntax errors that existed for a while

jina2k committed:
uninstall awesome-typescript-loader and some syntax spacing

jina2k committed:
removed most componentWill commands, warnings still appear

jina2k committed:
react-intl v2 to v5

jina2k committed:
aligning with dev

jina2k committed:
trying to remove conflicts before merge

jina2k committed:
merged packageupdate2

jina2k committed:
Merge branch 'development' into development

jina2k committed:
successful run with merge after reinstall

jina2k committed:
Merge branch 'development' of https://github.com/jina2k/OED into development

jina2k committed:
working

jina2k committed:
everything but react-intl v2 to v5

jina2k committed:
lint check and revert docker change

jina2k committed:
Merge branch 'development' into otherdev

jina2k committed:
react intl v2 to v5 and revert ts to 4.4.4

jina2k committed:
removed warnings

jina2k committed:
react-plotlyjs-ts somewhat deprecated, replaced with react/plotly.js

jina2k commented:
Figured I might as well attempt updating react, react-redux, redux-thunk, etc. as well as migrating to eslint.

For now, I have replaced react-plotlyjs-ts with react/plotly.js. (As of currently, size is set to width 100% using a parent css file as you cannot set width to 100% directly in plotly. height is set to 900px for line and bar)

This package was changed as I was attempting to remove react-plotlyjs-ts due to peer dependencies. Also linked is some issues revolving around this old package: https://github.com/davidctj/react-plotlyjs-ts/issues/10
Therefore, I followed the documentation provided here: https://plotly.com/javascript/react/


### Summary is as follows:
@babel/core@7.15.8

uninstalled react-plotlyjs-ts due to peer deps: plotly.js@1.31.2 and typescript@2.5.3
acorn-import-assertions@1.8.0 -> npm install --save-dev acorn@8

installed react-plotly.js instead
affected files:
BarChartContainer
CompareChartContainer
LineChartContainer
MapChartContainer
MapCalibrationChartDisplayContainer
DashboardComponent

config file tsconfig.json and declaration file desc.d.ts

jina2k committed:
static size of 900 for height, width set to 100% for plotly, height is only changed for both line and bar

jina2k committed:
Merge remote-tracking branch 'origin' into otherdev

jina2k committed:
wip of eslint

jina2k committed:
changing tab rules and removing semicolon rule

jina2k commented:
As seen in the most recent commit, there are multiple errors that are involved with eslint, ie; mostly commonjs vs es6 definitions. I haven't changed much regarding this, until further discussion is made. 
2 other big errors that I see are specifying types that are explicitly set to any, as well as tabs.
I have already changed rules for tabs, but there are still some files that do not align with these current rules. 
Another minor error is line length, as eslint counts line length differently from tslint. 

edit: if i attempt to include commonjs and es6/2021, as well as include option tabWidth set to 0 for max-len, error count# is 1105,154 for 1259 total (after fixing some tab spacing)

jina2k committed:
added commonjs for now, fixed mostly tab spacing

jina2k committed:
Merge branch 'development' into otherdev

jina2k committed:
Revert "added commonjs for now, fixed mostly tab spacing"

This reverts commit 2ddb9499bcd05785fa07b5c8c63de6410963b793.

jina2k committed:
Revert "changing tab rules and removing semicolon rule"

This reverts commit 87903a1085dad5545973d0d49857b5af0c12fe7c.

jina2k committed:
Revert "wip of eslint"

This reverts commit b0c4995156361e443b200d242acf7c61aa2fa82e.

jina2k committed:
reverted eslint for now

jina2k commented:
Fixed lingering issue that lasted regardless of whether I updated using npm audit or eslint related stuff. (declared json file as js, exported variable so it would be imported "properly")

As for ESLint, I will work on a separate branch for this (might work on React-redux first), put off for now due to commonjs problems where variables are not seen as defined as well as other issues.


tldr; 
changes made include:
babel -> @babel/core
awesome-typescript-loader -> ts-loader
react-plotlyjs-ts -> react-plotly.js
react-intl v2 ->v5 (stays at typescript 4.4.4 due to some datetime errors)

and then some renaming of imports (ie; InjectedIntlProps -> WrappedComponentProps, using componentDid______ instead of componentWill________)

npm audit for the most recent packages that were installed while working on this PR.

jina2k committed:
json issues with undefined

jina2k committed:
npm audit for nanoid

huss commented on src/client/app/actions/groups.ts:
```diff
@@ -271,7 +271,7 @@ function submitNewGroup(group: t.GroupData): Thunk {
 			dispatch(markGroupInEditingNotSubmitted());
 			if (e.response !== undefined && e.response.status === 400) {
-				showErrorNotification(e.response.data.error);
+				showErrorNotification(translate('400'));
```
I'm unclear on why the new, fixed message is better than the old one that gave the specific error that happened. I can see it translates but does it potentially show less information?

jina2k committed:
fixed error when changing bar stacking twice

jina2k commented:
https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html
https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html

referenced these two and stackoverflow after seeing error about calling setState too many times. Essentially replaced all componentDidUpdate calls that used setState with getDerivedStateFromProps. tested and it removes the error. 

also edited some other files to pass the lint check and other tests. related files include bash scripts to exclude the newly made data.js

huss committed:
add MPL to data.js

This file is now JS so can add the header.
Remove exclusion from check since now fine.

huss committed:
Merge branch 'development' of https://github.com/OpenEnergyDashboard/OED into pr/jina2k/696

huss commented:
I just pushed changes for this:

- I fixed up MPL check for new data.js file. What was there was okay but not it does not skip the check.
- I merged in development since it was showing a merge conflict on data.js. The merge did not show any issues so unsure why it said this. Hopefully cleared up now on GitHub.

huss reviewed (approved):
I have tested the changes and all seems to run fine. I appreciate @jina2k willingness to take on this important task and track down all the needed changes to upgrade packages. I also appreciate their willingness to look into other items and fix them as part of this work.
